ui/src/app/job-list/table/table.component.ts

```diff
@@ -249,4 +261,58 @@ export class JobsTableComponent implements OnInit {
       }
     }
   }
+
+  openBulkChangeDialog(): void {
+    this.resetBulkLabelFieldDefaults();
+    // figure out default values for bulk fields; if all jobs have the same label value,
+    // set that to the defualt; otherwise, set default to boolean false
+    for (let j = 0; j < this.selection.selected.length; j++) {
+      for (let f = 0; f < this.bulkLabelFields.length; f++) {
+        const label = this.bulkLabelFields[f].field.replace('labels.', '');
+        const jobLabelValue = this.selection.selected[j].labels[label] || '';
+        if (this.bulkLabelFields[f]['default'] == null) {
```

Comment:
I'd recommend using `null` to indicate no default, then use a string for everything else (i.e. `""` even if null).

Reply:
`null` means the default hasn't been set to anything yet, `false` means there's multiple values, so no default, `""` means the default has so far been set to an empty string

Reply:
My issue is that you have a value of type `boolean|string|null`, which is pretty confusing in my opinion. When you're rendering the dialog, the only thing you care about is "default value" or "(multi)". It seems more natural to me to express (multi) as null here.

Reply:
(also: see my other comment suggesting how to rework this code block - I don't think using null to track the value is necessary)

Reply:
I understand your issue with `false` being a possible value for default value. Your suggested code is definitely much more compact, but I also find it a little hard to read through, so I'm going to poke at it for a bit.

Reply:
That is one possible bug that could result here (false being a default value), but in general I would recommend against using union types like this anywhere. It makes for more confusing code IMO.

No need to use my exact code, but I don't think using null as a third state is doing anything for you currently. You could just as easily directly compare against the label value of the first job. Alternatively initialize defaultValue to the first label value, iterate through, and if something disagrees set defaultValue to null and break.